client/js/components/StatsLeaderboard.vue

```diff
@@ -0,0 +1,359 @@
+      switch (score) {
+        case 5:
+          return '#AF9500'; // gold
+        case 4:
+          return '#B4B4B4'; // silver
+        case 3:
+          return '#6A3805'; // bronze
+        case 1:
+          return 'primary';
+        default:
+          return '#000';
+      }
+    },
+    colorForTotalScore(score) {
+      if (score === this.topTotalScores.first) {
+        return '#AF9500'; // gold
+      }
+      if (score === this.topTotalScores.second) {
+        return '#B4B4B4'; // silver
+      }
+      if (score === this.topTotalScores.third) {
+        return '#6A3805'; // bronze
+      }
+      if (score > 0) {
+        return 'primary';
+      }
+      return '#000';
+    },
+    colorForScoreByWeek(week, score) {
+      return week === 'total' ? this.colorForTotalScore(score) : this.colorForScore(score);
+    },
+    /**
+     * Returns an object for v-bind for testing attributes to identify table cell
+     * @example {'points-2': 'someUserName'} identifies someUserNames' week-2 points
+     * @param {String} username
+     * @param {int} weekNum Which week
+     */
+    dataAttribute(username, weekNum) {
+      const res = {};
+      const attributeName = `week-${weekNum}`;
+      res[attributeName] = username;
+      return res;
+    },
+    /**
+     * Returns concatenated usernames of all opponent's the specified player
+     * defeated in the specified week
+     * @param {string} username which player's defeated opponents to return
+     * @param {string} weekNum which week to analyze (use 'total' for the total)
+     */
+    playersBeaten(username, weekNum) {
+      const playerStats = this.season.rankings.find(player => player.username === username);
+      if (!playerStats) {
+        return '';
+      }
+      let playerMatches;
+      // Aggregate all matches if looking at total
+      if (weekNum === 'total') {
+        playerMatches = Object.entries(playerStats.matches).reduce((wins, [week, matches]) => {
+          return [...wins, ...matches];
+        }, []);
+        // Otherwise just show this week's matches
+      } else {
+        playerMatches = playerStats.matches[weekNum];
+      }
+      if (!playerMatches) {
+        return [];
+      }
+      return playerMatches
+        .filter(match => match.result === Result.WON)
+        .map(match => match.opponent)
+        .join(', ');
+    },
+    isCurrentPlayer(username) {
+      return username === this.$store.state.auth.username;
+    },
+    tableRowClass(item) {
+      return this.isCurrentPlayer(item.username) ? 'active-user-stats' : '';
+    },
+    // Compute rank from total score
+    rank(totalScore) {
+      return this.totalScoresSorted.indexOf(totalScore) + 1;
+    },
+  },
+};
+</script>
+
+<style scoped>
+::v-deep .active-user-stats {
+  background-color: var(--v-accent-lighten3);
+}
+::v-deep .v-select.fit {
+  width: min-content;
+}
+::v-deep .v-select.fit .v-select__selection--comma {
+  text-overflow: unset;
+}
```

Comment:
v-deep is used to customize the styles children of the dynamically injected vuetify components